packages/spacecat-shared-http-utils/src/auth/auth-wrapper.js

```
@@ -48,6 +53,44 @@ export function authWrapper(fn, opts = {}) {
       return new Response('Unauthorized', { status: 401 });
     }
 
+    if (!context.auth) {
+      context.auth = {
+        hasScopes: async (scopes) => {
+          // Pull the api-key from x-api-key header
+          const apiKeyFromHeader = context.pathInfo?.headers['x-api-key'];
+
+          if (!hasText(apiKeyFromHeader)) {
+            return new Response('Unauthorized', { status: 401 });
+          }
+
+          // Generate a hash of the API Key
+          const hashedKey = hashWithSHA256(apiKeyFromHeader);
+
+          if (!context.dataAccess) {
+            log.error('Data access required');
+            return new Response('Server error', { status: 500 });
```

This should throw a new error IMO.